src/battle-log.ts

```diff
@@ -291,6 +295,37 @@ class BattleLog {
 		lastNode.appendChild(document.createTextNode(' '));
 		lastNode.appendChild(button);
 	}
+	unlinkChatFrom(userid: ID) {
+		const classStart = 'chat chatmessage-' + userid + ' ';
+		const innerNodeList = this.innerElem.getElementsByTagName('a');
+		const preemptNodeList = this.preemptElem.getElementsByTagName('a');
```

Comment:
Also: can you benchmark this, vs iterating child elements matching `classStart`, and using `getElementsByTagName` on each matching element?

The latter is much more readable to me, so unless the former is noticeably faster, I'd prefer that method.

Reply:
I've switched it around